src/main/java/woowacourse/shoppingcart/application/CartItemService.java

```diff
@@ -0,0 +1,84 @@
+import woowacourse.shoppingcart.domain.Amount;
+import woowacourse.shoppingcart.domain.Cart;
+import woowacourse.shoppingcart.domain.CartItem;
+import woowacourse.shoppingcart.domain.Product;
+import woowacourse.shoppingcart.dto.cart.CartItemCreateRequest;
+import woowacourse.shoppingcart.dto.cart.CartItemDto;
+import woowacourse.shoppingcart.dto.cart.CartItemDtos;
+import woowacourse.shoppingcart.exception.ExistSameProductIdException;
+import woowacourse.shoppingcart.exception.NoSuchProductException;
+import woowacourse.shoppingcart.exception.OutOfStockException;
+
+@Service
+@Transactional(rollbackFor = Exception.class)
+public class CartItemService {
+
+    private final CartItemDao cartItemDao;
+    private final ProductDao productDao;
+
+    public CartItemService(final CartItemDao cartItemDao, final ProductDao productDao) {
+        this.cartItemDao = cartItemDao;
+        this.productDao = productDao;
+    }
+
+    public List<CartItemDto> findCartsByCustomerId(final Long customerId) {
+        List<CartItem> cartItems = cartItemDao.findCartItemsByCustomerId(customerId);
+
+        CartItemDtos cartItemDtos = new CartItemDtos();
+
+        for (final CartItem cartItem : cartItems) {
+            final Product product = productDao.getProductById(cartItem.getProductId());
+            cartItemDtos.add(new CartItemDto(product, cartItem.getCount()));
+        }
+
+        return cartItemDtos.getValues();
+    }
+
+    public Long addCartItem(final Long customerId, final CartItemCreateRequest request) {
+        if (cartItemDao.existIdByCustomerIdAndProductId(customerId, request.getProductId())) {
+            throw new ExistSameProductIdException();
+        }
+
+        Product foundProduct = productDao.getProductById(request.getProductId());
+        validateCount(foundProduct, request.getCount());
+
+        return cartItemDao.addCartItem(customerId, request);
+    }
+
+    private void validateCount(Product product, int count) {
+        if (product.isAvailable(new Amount(count))) {
+            return;
+        }
+
+        throw new OutOfStockException();
+    }
+
+    public void updateCount(final Long customerId, final Long productId, final int newCount) {
+        Product foundProduct = productDao.getProductById(productId);
+        validateCount(foundProduct, newCount);
+
+        cartItemDao.updateCount(customerId, productId, newCount);
+    }
+
+    public void deleteCartItem(final Long customerId, final Long productId) {
+        validateProductInCustomerCart(customerId, productId);
+        cartItemDao.deleteCartItemByCustomerIdAndProductId(customerId, productId);
+    }
+
+    private void validateProductInCustomerCart(final Long customerId, final Long productId) {
+        Cart cart = new Cart(cartItemDao.findCartItemsByCustomerId(customerId));
+
+        if (cart.containsProductId(productId)) {
+            return;
+        }
+
+        throw new NoSuchProductException("장바구니에 해당 상품이 존재하지 않습니다.");
```

Comment:
```suggestion
        if (!cart.containsProductId(productId)) {
            throw new NoSuchProductException("장바구니에 해당 상품이 존재하지 않습니다.");
        }
```
개인적으로 이런 스타일이 코드도 짧고, 가독성이 좋은 것 같습니다 🙂 (반영하지 않으셔도 됩니다)